packages/dd-trace/src/esm-hook.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const semver = require('semver')
+
+if (semver.satisfies(process.versions.node, '^12.20.0 || >=14.13.1')) {
+  const parse = require('module-details-from-path')
+
+  const iitm = require('import-in-the-middle')
+
+  module.exports = function esmHook (instrumentedModules, hookFn) {
+    iitm.addHook((name, namespace) => {
```

Comment:
If all of this is needed to get an equivalent functionality, should it go in IITM? Otherwise, maybe the logic should be in the plugins instead, since the plugins will have knowledge of whether ESM is used or not?